lib/identify.c

```diff
@@ -1102,18 +1179,30 @@ uint8_t *quirc_begin(struct quirc *q, int *w, int *h)
 	return q->image;
 }
 
-void quirc_end(struct quirc *q)
+void quirc_end_with_workarea(struct quirc *q,
+			void *workarea, size_t workarea_size)
 {
+	struct workarea wa;
 	int i;
 
 	uint8_t threshold = otsu(q);
 	pixels_setup(q, threshold);
 
+	wa.p = workarea;
+	wa.len = workarea_size;
+
 	for (i = 0; i < q->h; i++)
-		finder_scan(q, i);
+		finder_scan(q, i, &wa);
 
 	for (i = 0; i < q->num_capstones; i++)
-		test_grouping(q, i);
+		test_grouping(q, i, &wa);
+}
+
+void quirc_end(struct quirc *q)
+{
+	struct flood_fill_vars stack[FLOOD_FILL_MAX_DEPTH];
+
+	quirc_end_with_workarea(q, (void *)stack, sizeof(stack));
```

Comment:
nit: casting to `void *` here seems unnecessary.

Reply:
fixed. thank you